src/ride/controllers/create_ride_controller.py

```diff
@@ -50,14 +50,26 @@ def process(self):
             return failure_response("Creator does not exist")
         creator_person = Person.objects.get(id=creator)
 
-        # Create path and ride and return new ride with given fields
-        path = Path.objects.create(
+        # Create new path or retrieve existing path
+        path_exists = Path.objects.filter(
```

Comment:
Why not save this as the existing path using .first() rather than using .exists()? Then you would not need an else statement with duplicate code

Reply:
+1, I remember that there was a reason @alanna-zhou had but I don't recall; this way is compacter and more efficient.

Reply:
using .exists() is more efficient in django rather than checking if the filtered query set is None (refer to https://docs.djangoproject.com/en/4.1/ref/models/querysets/#exists)

using get() is better than doing filter.first() since get()'s purpose is for getting only one object

Reply:
Thank you for linking.
`… but not by a large degree (hence needing a large queryset for efficiency gains).`
We are ensuring that we have unique paths by filtering on the ids. This means we will have **at most** one. Only use this paradigm if we're working with large amounts of data.